src/ripple/app/main/Application.cpp

```diff
@@ -882,20 +882,44 @@ class ApplicationImp
     {
     }
 
-    //------------------------------------------------------------------------------
+    //--------------------------------------------------------------------------
 
-    void onDeadlineTimer (DeadlineTimer& timer) override
+    void setSweepTimer ()
     {
-        if (timer == m_entropyTimer)
+        // Only start the timer if waitHandlerCounter_ is not yet joined.
+        if (auto optionalCountedHandler = waitHandlerCounter_.wrap (
+            [this] (boost::system::error_code const& e)
+            {
+                if ((e.value() == boost::system::errc::success) &&
+                    (! m_jobQueue->isStopped()))
+                {
+                    m_jobQueue->addJob(
+                        jtSWEEP, "sweep", [this] (Job&) { doSweep(); });
+                }
+            }))
         {
-            crypto_prng().mix_entropy ();
-            return;
+            sweepTimer_.expires_from_now (
+                std::chrono::seconds {config_->getSize (siSweepInterval)});
+            sweepTimer_.async_wait (std::move (*optionalCountedHandler));
         }
+    }
 
-        if (timer == m_sweepTimer)
+    void setEntropyTimer ()
+    {
+        // Only start the timer if waitHandlerCounter_ is not yet joined.
+        if (auto optionalCountedHandler = waitHandlerCounter_.wrap (
+            [this] (boost::system::error_code const& e)
+            {
+                if (e.value() == boost::system::errc::success)
+                {
+                    crypto_prng().mix_entropy();
+                    setEntropyTimer();
+                }
+            }))
         {
-            m_jobQueue->addJob(
-                jtSWEEP, "sweep", [this] (Job&) { doSweep(); });
+            using namespace std::chrono_literals;
+            entropyTimer_.expires_from_now (5min);
```

Comment:
I don't have any information about the rates either. I only mentioned it because the sweep timer is configurable and this one is not. A weak argument might be made that varying the entropy timer could increase difficulty for an attack knowing the server's time (interval is a given). I don't know if that is a case or not so I suggest leaving as is.